src/Server/EmbeddedClient/IClientDescriptorSet.h

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <iostream>
+
+namespace DB
+{
+
+
+// Interface, which handles descriptor pairs, that could be attached to embedded client
```

Comment:
Did you mean "file descriptor pairs"? If so, please change `DescriptorSet` to `FileDescriptorSet` \+ change the virtual method names to reflect "file descriptors".